x/upgrade/tally_test.go

```diff
@@ -0,0 +1,249 @@
+	"github.com/stretchr/testify/require"
+
+	testutil "github.com/celestiaorg/celestia-app/test/util"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
+	"github.com/tendermint/tendermint/libs/log"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	tmversion "github.com/tendermint/tendermint/proto/tendermint/version"
+	tmdb "github.com/tendermint/tm-db"
+)
+
+func TestSignalQuorum(t *testing.T) {
+	upgradeKeeper, ctx, _ := setup(t)
+	require.Equal(t, types.DefaultSignalQuorum, upgradeKeeper.SignalQuorum(ctx))
+	require.EqualValues(t, 100, upgradeKeeper.GetVotingPowerThreshold(ctx).Int64())
+	newParams := types.DefaultParams()
+	newParams.SignalQuorum = types.MinSignalQuorum
+	upgradeKeeper.SetParams(ctx, newParams)
+	require.Equal(t, types.MinSignalQuorum, upgradeKeeper.SignalQuorum(ctx))
+	require.EqualValues(t, 80, upgradeKeeper.GetVotingPowerThreshold(ctx).Int64())
+}
+
+func TestSignalVersion(t *testing.T) {
+	upgradeKeeper, ctx, _ := setup(t)
+	goCtx := sdk.WrapSDKContext(ctx)
+	_, err := upgradeKeeper.SignalVersion(goCtx, &types.MsgSignalVersion{
+		ValidatorAddress: testutil.ValAddrs[0].String(),
+		Version:          0,
+	})
+	require.Error(t, err)
+	_, err = upgradeKeeper.SignalVersion(goCtx, &types.MsgSignalVersion{
+		ValidatorAddress: testutil.ValAddrs[0].String(),
+		Version:          3,
+	})
+	require.Error(t, err)
+
+	_, err = upgradeKeeper.SignalVersion(goCtx, &types.MsgSignalVersion{
+		ValidatorAddress: testutil.ValAddrs[4].String(),
+		Version:          2,
+	})
+	require.Error(t, err)
+
+	_, err = upgradeKeeper.SignalVersion(goCtx, &types.MsgSignalVersion{
+		ValidatorAddress: testutil.ValAddrs[0].String(),
+		Version:          2,
+	})
+	require.NoError(t, err)
+
+	res, err := upgradeKeeper.VersionTally(goCtx, &types.QueryVersionTallyRequest{
+		Version: 2,
+	})
+	require.NoError(t, err)
+	require.EqualValues(t, 30, res.VotingPower)
+	require.EqualValues(t, 100, res.Threshold)
+	require.EqualValues(t, 120, res.TotalVotingPower)
+}
+
+func TestTallyingLogic(t *testing.T) {
+	upgradeKeeper, ctx, mockStakingKeeper := setup(t)
+	goCtx := sdk.WrapSDKContext(ctx)
+	_, err := upgradeKeeper.SignalVersion(goCtx, &types.MsgSignalVersion{
+		ValidatorAddress: testutil.ValAddrs[0].String(),
+		Version:          0,
+	})
+	require.Error(t, err)
+	_, err = upgradeKeeper.SignalVersion(goCtx, &types.MsgSignalVersion{
+		ValidatorAddress: testutil.ValAddrs[0].String(),
+		Version:          3,
+	})
+	require.Error(t, err)
+
+	_, err = upgradeKeeper.SignalVersion(goCtx, &types.MsgSignalVersion{
+		ValidatorAddress: testutil.ValAddrs[0].String(),
+		Version:          2,
+	})
+	require.NoError(t, err)
+
+	res, err := upgradeKeeper.VersionTally(goCtx, &types.QueryVersionTallyRequest{
+		Version: 2,
+	})
+	require.NoError(t, err)
+	require.EqualValues(t, 30, res.VotingPower)
+	require.EqualValues(t, 100, res.Threshold)
```

Comment:
[nit] this usage of `Threshold` makes me realize that the variable name `ThresholdPower` may make the code clearer

Reply:
Yup I can modify that